src/snowflake/snowpark/modin/plugin/_internal/frame.py

```diff
@@ -163,6 +174,29 @@ def create(
                 from_pandas_label(name, num_levels=1)
                 for name in data_column_pandas_index_names
             ),
+            snowflake_quoted_identifier_to_data_type={
```

Comment:
let's add an assertion helper method for checking type inputs, e.g.,

- check the input type length
- check the values either is None or SnowparkPandasType